### DISCUSSION TITLE
How do you get the size of a blob?

### DISCUSSION THREAD

**Discussion by drauschenbach** [n0-computer/iroh#2814] [General]
I thought this logic would aggregate blob content lengths:

```rust
    let mut num_files = 0;
    let mut total_lengths = 0;
    let query = Query::key_prefix("my_prefix:").build();
    let mut entries = my_document.get_many(query).await?;
    while let Some(maybe_entry) = entries.next().await {
        if let Ok(entry) = maybe_entry {
            let hash = entry.content_hash();
            let status = iroh_node.blobs().status(hash).await?;
            if let Complete { size } = status {
                num_files += 1;
                total_lengths += size as usize;
            }
        }
    }
```

However, sizes returned are a small number such as 360 bytes, when my file sizes are around 30M.

**Comment by matheus23** [comment]
`iroh_node.blobs().status(hash).await?` is the right way to get the blob size.
We have [tests](https://github.com/n0-computer/iroh/blob/d29537da6fc07ff82b8d56ff5fae9fdef9445858/iroh/src/client/blobs.rs#L1367) to ensure this number is correct.

I also modified another iroh_docs test to make sure this is not an issue with some interaction between iroh-docs and iroh-blobs.

Can you try to confirm this is not an issue with how you're adding the files to your document?
